src/services/authRequest.ts

```diff
@@ -1,16 +1,16 @@
 import request from '@/utils/request';
 import { AuthCredentials } from '@/constants/types';
 
-const authRequest = async <T>(endpoint: string, credentials: AuthCredentials): Promise<T> => {
-  const { data } = await request<T>(`${process.env.NEXT_PUBLIC_API}/users/${endpoint}`, {
+const authRequest = async (endpoint: string, credentials: AuthCredentials) => {
+  const { data, response } = await request(`${process.env.NEXT_PUBLIC_API}/users/${endpoint}`, {
     method: 'POST',
     body: JSON.stringify(credentials),
```

Comment:
use
```
if (config.body) {
  params.body = JSON.stringify(config.body)
}

```

or something similar in the request file, so you wouldn't need to write JSON.stringify every time.